pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -284,6 +284,38 @@ func BuildSidecarOutboundVirtualHosts(node *model.Proxy, push *model.PushContext
 	vhdomains := sets.Set{}
 	knownFQDN := sets.Set{}
 
+	buildVirtualHost := func(hostname string, vhwrapper istio_route.VirtualHostWrapper, svc *model.Service) *route.VirtualHost {
+		name := domainName(hostname, vhwrapper.Port)
+		if duplicateVirtualHost(name, vhosts) {
+			// This means this virtual host has caused duplicate virtual host name.
+			push.AddMetric(model.DuplicatedDomains, name, node.ID, fmt.Sprintf("duplicate domain from virtual service: %s", name))
```

Comment:
nit: before it was "from service" in the second case, here its always "virtual service"

Reply:
Yeah. Good catch. Fixed now